stable/admin/templates/job.yaml

```diff
@@ -22,7 +22,9 @@ spec:
         release: {{ .Release.Name | quote }}
       name: job-lb-policy-{{ .Values.admin.lbPolicy | lower }}
     spec:
+      {{- if .Values.nuodb.serviceAccount }}
```

Comment:
Is it sufficient to specify `--set nuodb.addServiceAccount=false` or do we also have to specify `--set nuodb.serviceAccount=''` to omit the service account name?

Reply:
Following up on this, I can't tell if the tests actually cover this change. If `nuodb.serviceAccount` is explicitly specified as the empty string, and the default value of `serviceAccountName` is the empty string, then it doesn't seem like we need to conditionalize this, unless we are actually trying to suppress the injection of `nuodb` as a default, in which case, wouldn't the following by better?

```
  serviceAccountName: {{ default "" .Values.nuodb.serviceAccount }}
```

Reply:
> Is it sufficient to specify --set nuodb.addServiceAccount=false or do we also have to specify --set nuodb.serviceAccount='' to omit the service account name?

We would like to support the following use cases:
1. Automatically create a service account (the default). `nuodb.addServiceAccount=true` and `nuodb.serviceAccount='nuodb'`
2. Don't create the service account but use preprovisioned one.  `nuodb.addServiceAccount=false` and `nuodb.serviceAccount='foo'`
3.  Don't create the service account and use the default one. `nuodb.addServiceAccount=false` and (`nuodb.serviceAccount=''` or `nuodb.serviceAccount=null` or `nuodb.serviceAccount='default'`)


> If nuodb.serviceAccount is explicitly specified as the empty string, and the default value of serviceAccountName is the empty string, then it doesn't seem like we need to conditionalize this, unless we are actually trying to suppress the injection of nuodb as a default, in which case, wouldn't the following by better (with no {{ if }} block conditionalizing it)?

This is an interesting question. It looks like if the `serviceAccountName` is left empty in the Pod template, Kubernetes will inject the `default` SA at runtime. So your proposal will work and it won't change the default behaviour because the `nuodb.serviceAccount=nuodb` is set as default in the `values.yaml` provided with the charts.